File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/ArticleListView.java
package com.phicdy.mycuration.presentation.view;

import android.support.annotation.NonNull;

public interface ArticleListView {
    void openInternalWebView(@NonNull String url, @NonNull String rssTitle);
    void openExternalWebView(@NonNull String url);
    void notifyListView();
    void finish();
    int getFirstVisiblePosition();
    int getLastVisiblePosition();
    void showShareUi(@NonNull String url);
    void scrollTo(int position);
    boolean isBottomVisible();
    void showEmptyView();
}

File: mycuration/src/main/java/com/phicdy/mycuration/presentation/view/ArticleListView.kt
package com.phicdy.mycuration.presentation.view

interface ArticleListView {
    val firstVisiblePosition: Int
    val lastVisiblePosition: Int
    val isBottomVisible: Boolean
    fun openInternalWebView(url: String, rssTitle: String)
    fun openExternalWebView(url: String)
    fun notifyListView()
    fun finish()
    fun showShareUi(url: String)
    fun scrollTo(position: Int)
    fun showEmptyView()
    fun showNoSearchResult()
}
